interfaces/systemd/spec.go

```diff
@@ -26,35 +26,51 @@ import (
 	"github.com/snapcore/snapd/snap"
 )
 
+type addedService struct {
+	iface string
+	svc   *Service
+}
+
 // Specification assists in collecting custom systemd services associated with an interface.
 //
 // Unlike the Backend itself (which is stateless and non-persistent) this type
-// holds internal state that is used by the mount backend during the interface
+// holds internal state that is used by the systemd backend during the interface
 // setup process.
 type Specification struct {
-	services map[string]*Service
+	curIface string
+	services map[string]*addedService
 }
 
 // AddService adds a new systemd service unit.
-func (spec *Specification) AddService(name string, s *Service) error {
-	if old, ok := spec.services[name]; ok && old != nil && s != nil && *old != *s {
-		return fmt.Errorf("interface requires conflicting system needs: service %q used to be defined as %#v, now re-defined as %#v", name, *old, *s)
+// distinctServiceAffix is used to name the service and needs to be unique.
```

Comment:
I don't know if it's obvious, but I'd add "(per snap)" at the end of this sentence.

Reply:
that sounded a bit ambiguous to me. What I did is adding this comment:

```
// Uniqueness across snaps is taken care implicitly elsewhere.
```